activestorage/lib/active_storage/service.rb

```diff
@@ -59,11 +60,19 @@ def configure(service_name, configurations)
       # Passes the configurator and all of the service's config as keyword args.
       #
       # See MirrorService for an example.
-      def build(configurator:, name:, service: nil, **service_config) # :nodoc:
+      def build(configurator:, name:, service: nil, url_config: nil, **service_config) # :nodoc:
+        if url_config
+          service_config = options_from_url(url_config)
```

Comment:
I know I suggested `options_from_url`, but looking at it in context, it should clearly be `service_config_from_url` to match the existing domain language. Also, would just go with `url` instead of `url_config`.